ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivacyApiMethod.java

```diff
@@ -0,0 +1,29 @@
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.privacy.methods.priv;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.JsonRpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.core.PrivacyParameters;
+
+public abstract class PrivacyApiMethod implements JsonRpcMethod {
+
+  protected final PrivacyParameters privacyParameters;
+
+  protected PrivacyApiMethod(final PrivacyParameters privacyParameters) {
+    this.privacyParameters = privacyParameters;
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
```

Comment:
Following this pattern you'll probably want to add the `final` modifier to `response()` to prevent any subclasses from overriding (prevent bypassing the enablement check).

Reply:
Nice catch!